### DISCUSSION TITLE
how to rotate the first caspule of cable in mujoco model

### DISCUSSION THREAD

**Discussion by hengyu2021** [google-deepmind/mujoco#2017] [Asking for Help]
I am trying to wrap a cable around the cylinder, this goal is achieved by giving the first capsule on the rope, the appropriate speed and rotation. But there is a problem in setting the rotation, I try to change the rotation angle of the joint, but it doesn't rotate to the angle I want.

for translation I set 
self.data.ctrl[11] =  vnew[0]
self.data.ctrl[12] =  vnew[1]
and it works good. But if I further set
self.data.ctrl[13] =  np.pi
It doesnot work. 

Here is a model which explains my question:
<mujoco>
  <compiler   autolimits="true" angle="radian" meshdir="./mesh" texturedir="./texture" />
  <extension>
    <plugin plugin="mujoco.elasticity.cable"/>    
  </extension>
  <statistic center="0 0 0.4" extent="1"/>
  <visual>
    <headlight diffuse="0.6 0.6 0.6" ambient="0.3 0.3 0.3" specular="0 0 0"/>
    <rgba haze="0.15 0.25 0.35 1"/>
    <global azimuth="120" elevation="-20"/>
  </visual>
  <option timestep="0.001" integrator="Euler" iterations="50" solver="Newton" impratio="5" noslip_iterations="5" tolerance="1e-8" cone="elliptic">
    <flag actuation="enable" gravity="enable" warmstart="enable" />
  </option>
  <asset>
    <texture type="skybox" builtin="gradient" rgb1="0.3 0.5 0.7" rgb2="0 0 0" width="512" height="3072"/>
    <texture type="2d" name="plane" builtin="checker" mark="edge" rgb1="0.2 0.3 0.4" rgb2="0.1 0.2 0.3"
      markrgb="0.8 0.8 0.8" width="300" height="300"/>
    <material name="plane" texture="plane" texuniform="true" texrepeat="5 5" reflectance="0.2"/>
  </asset>
  <include file="panda.xml" />
  <include file="surrounding/surr.xml"/>
  <worldbody>

    <body>      
      <geom type="box" size=".011 .011 .011" pos="-0.58 -0.23 0.33"  condim="6" friction="0.6 0.6 0.6"  />
      <body name="CB0" pos="-0.58 -0.23 0.13" euler="0 0 0">
        <joint name="pin" type="hinge" axis="0 0 1"  damping="1" />
        <joint name="pin1" type="hinge" axis="0 1 0"  damping="1" />
        <joint name="pin2" type="hinge" axis="1 0 0"  damping="1" />
        <joint name="sliderrrr" type="slide" axis="1 0 0"/>
        <joint name="sliderrrr1" type="slide" axis="0 1 0"/> 
        <composite type="cable" curve="s" count="50 1 1" size="1" offset="0 0 0" initial="none">  
          <plugin plugin="mujoco.elasticity.cable"> 
             
            <config key="twist" value="1e7"/>  
            <config key="bend" value="4e6"/>   
            <config key="vmax" value="0.05"/>  
          </plugin>
          <joint kind="main" damping=".015"/> 
          <geom type="capsule" size=".005" rgba=".8 .2 .1 1" condim="1"/>  
        </composite>
      </body>
    </body>
    <body name="target" pos="0 -0.45 0.2" euler="0 0 0">
      <geom name="tar" type="cylinder" size="0.1 0.1" rgba="0 .7 0 0.3"/>
    </body>
    <camera name="cam" mode="targetbody" pos="-2 -1 0.8" target="panda_hand" />
    <light directional="true" diffuse=".2 .2 .2" specular="0 0 0" pos="0 1 5" dir="0 -1 -1" castshadow="false"/>
    <light directional="false" diffuse=".8 .8 .8" specular="0.3 0.3 0.3" pos="0 -1 4" dir="0 0 -1" />
    <light directional="true" diffuse="0 0 0" specular=".7 .7 .7" pos="0 3 3" dir="0 -3 -3" />
    <geom name="floor" pos="0 0 0" group="1" size="0 0 .1" type="plane" material="plane" condim="3" />
  </worldbody>
  <equality />
  <actuator>
    <velocity name="slide" joint="sliderrrr" gear="1"  ctrllimited="true" ctrlrange="-6 6"/>
    <velocity name="slide1" joint="sliderrrr1" gear="1"  ctrllimited="true" ctrlrange="-6 6"/> 
    <position name="rotation" joint="pin" gear="1"  ctrllimited="true" ctrlrange="-1.571 1.571"/>
    <position name="rotation1" joint="pin1" gear="1"  ctrllimited="true" ctrlrange="-1.571 1.571"/>
    <position name="rotation2" joint="pin2" gear="1"  ctrllimited="true" ctrlrange="-1.571 1.571"/>
  </actuator>
</mujoco>

Here is a screenshot / video, illustrating my question:

https://github.com/google-deepmind/mujoco/assets/83298044/40850bbb-6c3c-4650-90f1-54f67eef019a

**Comment by quagla** [comment]
Are you sure about your indexing? What type of actuator are you using?

**Comment by hengyu2021** [comment]
> Are you sure about your indexing? What type of actuator are you using?
I am sure about the indexing, and the actuator is defined as following

     <velocity name="slide" joint="sliderrrr" gear="1"  ctrllimited="true" ctrlrange="-6 6"/>
    <velocity name="slide1" joint="sliderrrr1" gear="1"  ctrllimited="true" ctrlrange="-6 6"/> 
    <position name="rotation" joint="pin" gear="1"  ctrllimited="true" ctrlrange="-1.571 1.571"/>
    <position name="rotation1" joint="pin1" gear="1"  ctrllimited="true" ctrlrange="-1.571 1.571"/>
    <position name="rotation2" joint="pin2" gear="1"  ctrllimited="true" ctrlrange="-1.571 1.571"/>
